### PR TITLE
Add cluster hibernation support

### PR DESCRIPTION
**Pull request by nimrodshn**
cc @igoihman

### CODE REVIEW

**Comment by nimrodshn** [comment]
@igoihman Please note that the current API is not supported using the ocm-model / sdk and @jhernand  suggested that we change our "resume from hibernation" / "unhibernate API from :`DELETE /api/clusters_mgmt/v1/clusters/<id>/actions/hibernate` to `POST /api/clusters_mgmt/v1/clusters/<id>/actions/unhibernate`

@jhernand  had also suggested we drop the `/actions` preface, 

@tzvatot @igoihman WDYT?

**Comment by igoihman** [comment]
@nimrodshn let's drop the `/actions` preface. The suggested approach sounds good, what about calling the new endpoint `resume`?

File: model/clusters_mgmt/v1/cluster_resource.model [hunk 1]
```diff
@@ -75,6 +75,17 @@ resource Cluster {
 		target MachinePools
 	}
 
+	// Initiates cluster hibernation; While hibernating a cluster will not consume any cloud provider infrastructure
+	// but will be counted for quota.
```
Comment (jhernand) [inline model/clusters_mgmt/v1/cluster_resource.model:79]: This documentation isn't currently used for anything other than putting it in the OpenAPI documentation. But in the future I would like to use it to generate human readable documentation. In order to do that I would like to have in each piece of documentation a short sentence describing the thing. Then a dot, and then possible additional, longer, documentation. That dot will be used by the tool that generates the human readable documentation. So, can you reformat like this?

```
// Initiates cluster hibernation. While hibernating a cluster will not consume any cloud provider infrastructure
// but will be counted for quota.
```

File: model/clusters_mgmt/v1/cluster_resource.model [hunk 2]
```diff
@@ -75,6 +75,17 @@ resource Cluster {
...
+    method Hibernate {
```
Comment (jhernand) [inline model/clusters_mgmt/v1/cluster_resource.model:80]: Make sure to use tabs instead of spaces for indentation.

**Review by jhernand — COMMENTED** [review]
Looks good. Just some minor formatting comments. Address them an I will merge.

File: model/clusters_mgmt/v1/cluster_resource.model (hunk 2 above)
Comment (nimrodshn) [reply model/clusters_mgmt/v1/cluster_resource.model:80]: Done. thanks

File: model/clusters_mgmt/v1/cluster_resource.model (hunk 1 above)
Comment (nimrodshn) [reply model/clusters_mgmt/v1/cluster_resource.model:79]: Thanks, done.

**Review by jhernand — APPROVED** [review]
Looks good, thanks.

**Pull request merged**